### PR TITLE
Add support for `exit_group` syscall

### PR DESCRIPTION
**Pull request by rvolosatovs**
Part of #34

### CODE REVIEW

File: src/v2/src/host/syscall.rs [hunk 1]
```diff
@@ -148,6 +148,17 @@ pub(super) unsafe fn execute_syscall(syscall: &mut item::Syscall, data: &mut [u8
         }
         .execute(),
 
+        item::Syscall {
+            num,
+            argv: [status, ..],
+            ret: [ret, ..],
+        } if *num == libc::SYS_exit_group as _ => Syscall {
+            num: libc::SYS_exit_group,
```
Comment (jarkkojs) [inline src/v2/src/host/syscall.rs:156]: Not something I would necessarily NAK the commit, but I would have probably added a small inline comment denoting that glibc does not wrap this. Anyway, I'm fine as it is too...

File: src/v2/src/guest/handler.rs [hunk 2]
```diff
@@ -54,6 +54,9 @@ pub trait Execute {
         match (num as _, argv) {
             (libc::SYS_close, [fd, ..]) => self.close(fd as _).map(|_| [0, 0]),
             (libc::SYS_exit, [status, ..]) => self.exit(status as _).map(|_| self.attacked()),
+            (libc::SYS_exit_group, [status, ..]) => {
+                self.exit_group(status as _).map(|_| self.attacked())
```
Comment (jarkkojs) [inline src/v2/src/guest/handler.rs:58]: What is self.attacked()?

File: src/v2/src/host/syscall.rs (hunk 1 above)
Comment (rvolosatovs) [reply src/v2/src/host/syscall.rs:156]: We don't have access to standard library and, hence, glibc in this package. This particular piece of code matches the requested syscall read from the untrusted block and executes it via assembly instruction directly akin to libc implementations like glibc, musl etc.
I'm not entirely sure what you suggest documenting.

File: src/v2/src/guest/handler.rs (hunk 2 above)
Comment (rvolosatovs) [reply src/v2/src/guest/handler.rs:58]: It tries to exit in an infinite loop - it's called in cases when the guest determines that the integrity of the shared block was compromised, e.g. if there were more bytes read via `read`, than requested, or, in this case, control was returned to the guest after an `exit_group`, which was successfully proxied to the host

File: src/v2/src/host/syscall.rs (hunk 1 above)
Comment (jarkkojs) [reply src/v2/src/host/syscall.rs:156]: Got it thanks!

File: src/v2/src/guest/handler.rs (hunk 2 above)
Comment (jarkkojs) [reply src/v2/src/guest/handler.rs:58]: Thank you

**Comment by jarkkojs** [comment]
How do I approve this? I understand the code change now and it is legit.

**Review by haraldh — APPROVED** [review]

**Pull request merged**